### ISSUE TITLE
InternalException when nesting window functions via an alias reference

### ISSUE THREAD

**Issue by dunnkevin** [duckdb/duckdb#8462]
### What happens?

When running a query that nests window functions, an BinderException is thrown indicating nesting window functions is not allowed. However, if the nested window function is in the same select statement and referenced via an alias, an InternalException is thrown with a failed to bind column error, and the database is invalidated.

### To Reproduce

Run the following:

```
import duckdb

query = """
select
'apple' as str,
sum(1) over (partition by str) as wf1,
sum(wf1) over (partition by str) as wf2
"""

with duckdb.connect(database=':memory:') as conn:
    conn.execute(query)
```

Results in:
`duckdb.duckdb.InternalException: INTERNAL Error: Failed to bind column reference "wf1" [5.0] (bindings: [0.0])`

### OS:

MacOS, Linux

### DuckDB Version:

0.8.2.dev2356

### DuckDB Client:

Python

### Full Name:

Kevin Dunn

### Affiliation:

Hex Technologies

### Have you tried this on the latest `master` branch?

I have tested with a master build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by hawkfish** [comment]
This happens in the binding code a long way from the window planner.

It raises an interesting question, however, in that the window planner assume it can reorder computations arbitrarily for performance. If they depend on each other then either we have to generate multiple `LogicalWindowOperator`s or the planner needs to build a dependency graph and throw that into the mix (effectively partitioning the single logical operator).

**Comment by Mytherin** [comment]
Fixed in v0.9.2

```sql
select
'apple' as str,
sum(1) over (partition by str) as wf1,
sum(wf1) over (partition by str) as wf2
-- Error: Binder Error: window function calls cannot be nested

```
